Dependency cycles between libs

Seem like a recent-ish regression
```
$ ninja -k0 -j1
[1/3] Linking CXX executable inst_tests
FAILED: inst_tests 
: && /usr/local/bin/clang++  -g -ggdb -UNDEBUG -O3 -g -UNDEBUG  -rdynamic CMakeFiles/inst_tests.dir/unittests/Inst/InstTests.cpp.o  -o inst_tests  -Wl,-rpath,/opt/souper/third_party/alive2/build libsouperInfer.a libsouperInst.a -lgtest_main -lgtest ../third_party/alive2/build/libalive2.so ../third_party/z3-install/lib/libz3.a -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoDWARF -lLLVMFuzzMutate -lLLVMTableGen -lLLVMDlltoolDriver -lLLVMLineEditor -lLLVMOrcJIT -lLLVMCoverage -lLLVMMIRParser -lLLVMObjectYAML -lLLVMLibDriver -lLLVMOption -lLLVMWindowsManifest -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMGlobalISel -lLLVMSelectionDAG -lLLVMAsmPrinter -lLLVMX86Desc -lLLVMMCDisassembler -lLLVMX86Info -lLLVMX86AsmPrinter -lLLVMX86Utils -lLLVMMCJIT -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMCodeGen -lLLVMTarget -lLLVMCoroutines -lLLVMipo -lLLVMInstrumentation -lLLVMVectorize -lLLVMScalarOpts -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMInstCombine -lLLVMBitWriter -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMAnalysis -lLLVMProfileData -lLLVMObject -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMDebugInfoMSF -lLLVMBitReader -lLLVMCore -lLLVMBinaryFormat -lLLVMSupport -lLLVMDemangle -lz -lrt -ldl -ltinfo -lpthread -lm -lxml2 -L/opt/souper/third_party/llvm/Release/lib -L /opt/souper/third_party/alive2/build -lalive2 && :
ld: error: undefined symbol: souper::BuildQuery[abi:cxx11](souper::InstContext&, std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::InstMapping, std::vector<souper::Inst*, std::allocator<souper::Inst*> >*, bool)
>>> referenced by InstSynthesis.cpp:234 (../lib/Infer/InstSynthesis.cpp:234)
>>>               InstSynthesis.cpp.o:(souper::InstSynthesis::synthesize(souper::SMTLIBSolver*, std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&, unsigned int)) in archive libsouperInfer.a

ld: error: undefined symbol: souper::BuildQuery[abi:cxx11](souper::InstContext&, std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::InstMapping, std::vector<souper::Inst*, std::allocator<souper::Inst*> >*, bool)
>>> referenced by InstSynthesis.cpp:177 (../lib/Infer/InstSynthesis.cpp:177)
>>>               InstSynthesis.cpp.o:(souper::InstSynthesis::synthesize(souper::SMTLIBSolver*, std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&, unsigned int)) in archive libsouperInfer.a

ld: error: undefined symbol: souper::BuildQuery[abi:cxx11](souper::InstContext&, std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::InstMapping, std::vector<souper::Inst*, std::allocator<souper::Inst*> >*, bool)
>>> referenced by InstSynthesis.cpp:1362 (../lib/Infer/InstSynthesis.cpp:1362)
>>>               InstSynthesis.cpp.o:(souper::InstSynthesis::getInitialConcreteInputs(std::vector<std::map<souper::Inst*, souper::Inst*, std::less<souper::Inst*>, std::allocator<std::pair<souper::Inst* const, souper::Inst*> > >, std::allocator<std::map<souper::Inst*, souper::Inst*, std::less<souper::Inst*>, std::allocator<std::pair<souper::Inst* const, souper::Inst*> > > > >&, unsigned int)) in archive libsouperInfer.a
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[2/3] Linking CXX executable extractor_tests
FAILED: extractor_tests 
: && /usr/local/bin/clang++  -g -ggdb -UNDEBUG -O3 -g -UNDEBUG  -rdynamic CMakeFiles/extractor_tests.dir/unittests/Extractor/ExtractorTests.cpp.o  -o extractor_tests  -Wl,-rpath,/opt/souper/third_party/alive2/build libsouperExtractor.a -lgtest_main -lgtest ../third_party/alive2/build/libalive2.so libsouperInfer.a ../third_party/z3-install/lib/libz3.a libsouperInst.a libkleeExpr.a -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoDWARF -lLLVMFuzzMutate -lLLVMTableGen -lLLVMDlltoolDriver -lLLVMLineEditor -lLLVMOrcJIT -lLLVMCoverage -lLLVMMIRParser -lLLVMObjectYAML -lLLVMLibDriver -lLLVMOption -lLLVMWindowsManifest -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMGlobalISel -lLLVMSelectionDAG -lLLVMAsmPrinter -lLLVMX86Desc -lLLVMMCDisassembler -lLLVMX86Info -lLLVMX86AsmPrinter -lLLVMX86Utils -lLLVMMCJIT -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMCodeGen -lLLVMTarget -lLLVMCoroutines -lLLVMipo -lLLVMInstrumentation -lLLVMVectorize -lLLVMScalarOpts -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMInstCombine -lLLVMBitWriter -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMAnalysis -lLLVMProfileData -lLLVMObject -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMDebugInfoMSF -lLLVMBitReader -lLLVMCore -lLLVMBinaryFormat -lLLVMSupport -lLLVMDemangle -lz -lrt -ldl -ltinfo -lpthread -lm -lxml2 -L/opt/souper/third_party/llvm/Release/lib -L /opt/souper/third_party/alive2/build -lalive2 && :
ld: error: undefined symbol: souper::ParseReplacementRHS(souper::InstContext&, llvm::StringRef, llvm::StringRef, souper::ReplacementContext&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
>>> referenced by Solver.cpp:344 (../lib/Extractor/Solver.cpp:344)
>>>               Solver.cpp.o:((anonymous namespace)::MemCachingSolver::infer(std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&)) in archive libsouperExtractor.a

ld: error: undefined symbol: souper::KVStore::hGet(llvm::StringRef, llvm::StringRef, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
>>> referenced by Solver.cpp:400 (../lib/Extractor/Solver.cpp:400)
>>>               Solver.cpp.o:((anonymous namespace)::ExternalCachingSolver::infer(std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&)) in archive libsouperExtractor.a

ld: error: undefined symbol: souper::ParseReplacementRHS(souper::InstContext&, llvm::StringRef, llvm::StringRef, souper::ReplacementContext&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
>>> referenced by Solver.cpp:406 (../lib/Extractor/Solver.cpp:406)
>>>               Solver.cpp.o:((anonymous namespace)::ExternalCachingSolver::infer(std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&)) in archive libsouperExtractor.a

ld: error: undefined symbol: souper::KVStore::hSet(llvm::StringRef, llvm::StringRef, llvm::StringRef)
>>> referenced by Solver.cpp:416 (../lib/Extractor/Solver.cpp:416)
>>>               Solver.cpp.o:((anonymous namespace)::ExternalCachingSolver::infer(std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&)) in archive libsouperExtractor.a

ld: error: undefined symbol: souper::KVStore::hSet(llvm::StringRef, llvm::StringRef, llvm::StringRef)
>>> referenced by Solver.cpp:424 (../lib/Extractor/Solver.cpp:424)
>>>               Solver.cpp.o:((anonymous namespace)::ExternalCachingSolver::infer(std::vector<souper::BlockPCMapping, std::allocator<souper::BlockPCMapping> > const&, std::vector<souper::InstMapping, std::allocator<souper::InstMapping> > const&, souper::Inst*, souper::Inst*&, souper::InstContext&)) in archive libsouperExtractor.a
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[3/3] Building souperweb
ninja: build stopped: cannot make progress due to previous errors.
```
Also, `CMakeLists.txt` is a mess, i'm not sure i can fix this issue without reshuffling it all..

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -345,8 +345,8 @@ target_link_libraries(souper-check souperTool souperExtractor souperKVStore soup
 target_link_libraries(souper-interpret souperTool souperExtractor souperKVStore souperSMTLIB2 souperParser ${HIREDIS_LIBRARY} ${ALIVE_LIBRARY} z3)
 target_link_libraries(clang-souper souperClangTool souperExtractor souperKVStore souperParser souperSMTLIB2 souperTool kleeExpr ${CLANG_LIBS} ${LLVM_LIBS} ${LLVM_LDFLAGS} ${HIREDIS_LIBRARY} ${ALIVE_LIBRARY} z3)
 target_link_libraries(count-insts souperParser)
-target_link_libraries(extractor_tests souperExtractor ${GTEST_LIBS} ${ALIVE_LIBRARY})
-target_link_libraries(inst_tests souperInfer souperInst ${GTEST_LIBS} ${ALIVE_LIBRARY})
+target_link_libraries(extractor_tests souperExtractor souperKVStore ${HIREDIS_LIBRARY} souperParser ${GTEST_LIBS} ${ALIVE_LIBRARY})
+target_link_libraries(inst_tests souperInfer souperInst souperExtractor ${GTEST_LIBS} ${ALIVE_LIBRARY})
 target_link_libraries(parser_tests souperParser ${GTEST_LIBS} ${ALIVE_LIBRARY})
 
 SET(BUILD_CLANG_TOOL 1 CACHE BOOL "Build the Souper Clang tool")

